doc/neo-tree.txt

```diff
@@ -1208,6 +1208,52 @@ This will render:
     FILENAME.js
         FILENAME.js.map
 
+A more advanced usage are rules with patterns. Those can be mixed with file
```

Comment:
I'm not shure if I understand what exactly you mean. The patterns are regex-like Lua patterns (https://www.lua.org/pil/20.2.html) which can have captures (https://www.lua.org/pil/20.3.html). Via the captures we can check for parts of an filename to be used for nested files. e.g. in go, we can "capture" the filename and then determine, what file ist the test file.
If this is not what you meant please let me know.

Reply:
I guess my confusion is that I thought these were global patterns and not lua patterns, because you are using globtopattern to convert them. Now I see that these are actually lua patterns so the `%` I was confused about were actually escaping the `.` to be a literal dot.

If that is the case, why are you using globtopattern?

We normally use that library because globs are pretty universal and lua patterns are not, so most people would prefer to use globs rather than learn a new syntax.

Reply:
I wanted the pattern matching to be more or less universal but also as easy to use as possible. So for file matching of the "main" file I use patterns so I can have captures. Those captures can then be used for matching the nested files, but thos have to be globs in order to have wildcards for different matching rules. As globs are easier to use if there are many files to match and it's easier to use the captures without breaking patterns. Maybe this is a bit overenginieered or inconsistent but it works ;-)
I will try to explain the "flow" with an example: 

```lua
"test" = {
    pattern = "foo_(.*)%.bar"
    files = { "foobar.*", "%1.sub", "child_%1.*" }
}
```

The pattern would match a file `foo_test.bar` with a capture `test`. After we have the capture we can insert it into the file globs: `foobar.*`, `test.sub`, `child_test.*`. These globs are then converted to patterns to match files like `foobar.md`, `test.sub`, `child_test.js`, `child_test.tmp`

I already have a TODO on my list to think about checking the nested file pattern if this is obviously a glob or just a fixed string. If the latter we could skip the globtopattern part and just compare the file name. But first I have to refactor my code ;-)

If you think this is too clunky let me know if you would prefer a different approach - or if you dont't like it at all just reject the pull request - no hard feelings 😊

Reply:
OK, I get it. It is a definitely confusing that we are using two different types of patterns within the same table. I do understand why now. I don't have any better ideas, but I would say that if this is the only way then it needs to be spelled out very clearly that one is lua pattern and one is a glob pattern. I would include the explanation in the docs and also add a reminder anywhere an example is displayed. Like this:

``` lua
"test" = {
    pattern = "foo_(.*)%.bar", -- <-- Lua pattern
    files = { "foobar.*", "%1.sub", "child_%1.*" } -- <-- glob pattern
}
```